SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -95,152 +98,204 @@ void SpatialLoadBalanceEnforcer::MaybeQueueAclAssignmentRequest(const Worker_Ent
 		return;
 	}
 
-	const SpatialGDK::AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(EntityId);
-	const PhysicalWorkerName* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
-
-	check(OwningWorkerId != nullptr);
-	if (OwningWorkerId == nullptr)
-	{
-		UE_LOG(LogSpatialLoadBalanceEnforcer, Error, TEXT("Couldn't find mapped worker for entity %lld. This shouldn't happen! Virtual worker ID: %d"),
-			EntityId, AuthorityIntentComponent->VirtualWorkerId);
-		return;
-	}
-
-	if (*OwningWorkerId == WorkerId && StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
+	if (!EntityNeedsToBeEnforced(EntityId))
 	{
-		UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("No need to queue newly authoritative entity because this worker is already authoritative. Entity: %lld. Worker: %s."),
-			EntityId, *WorkerId);
 		return;
 	}
 
 	if (AclAssignmentRequestIsQueued(EntityId))
 	{
-		UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("Avoiding queueing a duplicate ACL assignment request. Entity: %lld. Worker: %s."),
-			EntityId, *WorkerId);
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("Avoiding queueing a duplicate ACL assignment request. "
+			"Entity: %lld. Worker: %s."), EntityId, *WorkerId);
 		return;
 	}
 
-	QueueAclAssignmentRequest(EntityId);
+	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Queueing ACL assignment request. Entity %lld. Worker %s."), EntityId, *WorkerId);
+	PendingEntityAuthorityChanges.Add(EntityId);
 }
 
-bool SpatialLoadBalanceEnforcer::AclAssignmentRequestIsQueued(const Worker_EntityId EntityId) const
+bool SpatialLoadBalanceEnforcer::EntityNeedsToBeEnforced(const Worker_EntityId EntityId) const
```

Comment:
Currently, we do a bunch of unnecesssary EntityACL updates when enforcing worker checks out components,  this function sanity checks whether any changes are actually necessary.